diff --git a/node_modules/next/dist/shared/lib/router/router.js b/node_modules/next/dist/shared/lib/router/router.js
index 1b0086f..0d30966 100644
--- a/node_modules/next/dist/shared/lib/router/router.js
+++ b/node_modules/next/dist/shared/lib/router/router.js
@@ -58,7 +58,11 @@
 if (process.env.__NEXT_I18N_SUPPORT) {
     detectDomainLocale = require('../i18n/detect-domain-locale').detectDomainLocale;
 }
-const basePath = process.env.__NEXT_ROUTER_BASEPATH || '';
+let basePath = process.env.__NEXT_ROUTER_BASEPATH || '';
+exports.setBasePath = function(newBasePath) {
+  basePath = newBasePath;
+};
+
 function buildCancellationError() {
     return Object.assign(new Error('Route Cancelled'), {
         cancelled: true
